test/e2e/app/app.go

```diff
@@ -456,10 +457,10 @@ func (app *Application) PrepareProposal(
 			app.logger.Error("detected tx that should not come from the mempool", "tx", tx)
 			continue
 		}
-		if totalBytes+int64(len(tx)) > req.MaxTxBytes {
+		if totalBytes+cmttypes.ComputeProtoSizeForTxs([]cmttypes.Tx{tx}) > req.MaxTxBytes {
```

Comment:
extract to variable to avoid double marshaling to proto?

Reply:
Good point. Fixed.